src/renderer/components/recent-files.js

```diff
@@ -1,34 +1,54 @@
 import React, { Component, PropTypes } from 'react';
 import DeleteIcon from 'react-icons/lib/ti/delete-outline';
 import HistoryIcon from 'react-icons/lib/go/history';
+import CloudIcon from 'react-icons/lib/md/cloud-queue';
+import DropboxIcon from 'react-icons/lib/fa/dropbox';
+import LaptopIcon from 'react-icons/lib/md/laptop-mac';
 import { Button } from 'buttercup-ui';
+import { archiveTypes } from '../../shared/buttercup/types';
 import styles from '../styles/recent-files';
 import { parsePath } from '../system/utils';
 import { showContextMenu } from '../system/menu';
+import EmptyView from './empty-view';
 
-const File = ({filename, onClick, onRemoveClick}) => {
-  const { base, dir } = parsePath(filename);
+function getIcon(type) {
+  switch (type) {
+    case archiveTypes.OWNCLOUD:
```

Comment:
Should we maybe keep these icons consistent? https://github.com/buttercup/buttercup-browser-extension/tree/master/resources/providers